transport.go

```diff
@@ -164,13 +186,22 @@ func (t *Transport) refreshToken(ctx context.Context) error {
 	t.appsTransport.BaseURL = t.BaseURL
 	t.appsTransport.Client = t.Client
 	resp, err := t.appsTransport.RoundTrip(req)
+	e := RefreshTokenError{
+		Message:        fmt.Sprintf("received non 2xx response status %q when fetching %v", resp.Status, req.URL),
+		RootCause:      err,
+		InstallationID: t.installationID,
+		Request:        req,
+		Response:       resp,
+	}
 	if err != nil {
-		return fmt.Errorf("could not get access_tokens from GitHub API for installation ID %v: %v", t.installationID, err)
+		e.Message = fmt.Sprintf("could not get access_tokens from GitHub API for installation ID %v: %v", t.installationID, err)
+		return e
 	}
 	defer resp.Body.Close()
 
 	if resp.StatusCode/100 != 2 {
-		return fmt.Errorf("received non 2xx response status %q when fetching %v", resp.Status, req.URL)
+		e.Message = fmt.Sprintf("received non 2xx response status %q when fetching %v", resp.Status, req.URL)
+		return e
 	}
 
 	return json.NewDecoder(resp.Body).Decode(&t.token)
```

Comment:
This is returning a non-RefreshTokenError error type. I wonder if we should return a RefreshTokenError here as well, though we would need to be careful about preserving the body for callers to consume. 🤔

Reply:
Good question. I am not sure as well about this.
On the one side, this would be consistent. On the other side, the `refreshToken` has also other cases where a non-RefreshTokenError error type is returned:
- Creating the request
- receiving a GetReadWriter

Semantically, not being able to decode the response, should return a RefreshTokenError.
About closing the body: Here, I would extend the documentation of `func (t *Transport) Token` by:

> If err is from type `RefreshTokenError`, the caller need to close the response body.

Let me know what you think and what the decision is. I will adjust the code.

Reply:
How about renaming the type to `HTTPError` or something similar to make it clear that this is debug information for HTTP requests? Then it might be clearer that mixed error types can be returned.

Reply:
Good suggestion. Fixed in https://github.com/bradleyfalzon/ghinstallation/pull/45/commits/92d39fffee5a87b78ebaf85e2a48aedb42086fee